test/discr/discr_ops_test.py

```diff
@@ -0,0 +1,181 @@
+
+    odl.ResizingOperator(space, res_space)
+    odl.ResizingOperator(space, ran_shp=(20, 15))
+    odl.ResizingOperator(space, ran_shp=(20, 15), num_left=(0, 5))
+    odl.ResizingOperator(space, ran_shp=(20, 15), pad_mode='symmetric')
+    odl.ResizingOperator(space, ran_shp=(20, 15), pad_const=1.0)
+    odl.ResizingOperator(space, ran_shp=(20, 15), pad_const=1.0,
+                         discr_kwargs={'nodes_on_bdry': True})
+
+
+def test_resizing_op_raise():
+
+    # domain not a uniformely discretized Lp
+    with pytest.raises(TypeError):
+        odl.ResizingOperator(odl.rn(5), ran_shp=(10,))
+
+    grid = odl.TensorGrid([0, 2, 3])
+    part = odl.RectPartition(odl.IntervalProd(0, 3), grid)
+    fspace = odl.FunctionSpace(odl.IntervalProd(0, 3))
+    dspace = odl.rn(3)
+    space = odl.DiscreteLp(fspace, part, dspace)
+    with pytest.raises(ValueError):
+        odl.ResizingOperator(space, ran_shp=(10,))
+
+    # different cell sides in domain and range
+    space = odl.uniform_discr(0, 1, 10)
+    res_space = odl.uniform_discr(0, 1, 15)
+    with pytest.raises(ValueError):
+        odl.ResizingOperator(space, res_space)
+
+    # non-integer multiple of cell sides used as shift (grid of the
+    # resized space shifted)
+    space = odl.uniform_discr(0, 1, 5)
+    res_space = odl.uniform_discr(-0.5, 1.5, 10)
+    with pytest.raises(ValueError):
+        odl.ResizingOperator(space, res_space)
+
+    # need either range or ran_shp
+    with pytest.raises(ValueError):
+        odl.ResizingOperator(space)
+
+    # num_left cannot be combined with range
+    space = odl.uniform_discr([0, -1], [1, 1], (10, 5))
+    res_space = odl.uniform_discr([0, -3], [2, 3], (20, 15))
+    with pytest.raises(ValueError):
+        odl.ResizingOperator(space, res_space, num_left=(0, 0))
+
+    # bad pad_mode
+    with pytest.raises(ValueError):
+        odl.ResizingOperator(space, res_space, pad_mode='something')
+
+
+def test_resizing_op_properties():
+
+    # Explicit range, rest default
+    space = odl.uniform_discr([0, -1], [1, 1], (10, 5))
+    res_space = odl.uniform_discr([0, -3], [2, 3], (20, 15))
+    res_op = odl.ResizingOperator(space, res_space)
+
+    assert res_op.domain == space
+    assert res_op.range == res_space
+    assert res_op.num_left == (0, 5)
+    assert res_op.pad_mode == 'constant'
+    assert res_op.pad_const == 0.0
+    assert res_op.is_linear
+
+    # Implicit range via ran_shp and num_left
+    res_op = odl.ResizingOperator(space, ran_shp=(20, 15), num_left=[0, 5])
+    assert res_op.range == res_space
+    assert res_op.num_left == (0, 5)
+
+    # Different padding mode
+    res_op = odl.ResizingOperator(space, res_space, pad_const=1)
+    assert res_op.pad_const == 1.0
+    assert not res_op.is_linear
+
+    res_op = odl.ResizingOperator(space, res_space, pad_mode='symmetric')
+    assert res_op.pad_mode == 'symmetric'
+    assert res_op.is_linear
+
+
+def test_resizing_op_call(fn_impl):
```

Comment:
This should have a fixture, and I.m.o should test much more corner cases (axis size 0 and 1, different dtypes, data orders, what happens if space is over integers? etc)